camayoc/tests/qpc/cli/test_openshift.py

```diff
@@ -77,6 +77,30 @@ def validate_openshift_report(cluster, details, deployments):
     assert set(cluster_nodes) == set(seen_nodes), "The number of expected nodes diverged"
 
 
+def validate_openshift_with_acm(cluster, details):
+    EXPECTED_METRICS_KEYS = (
+        "vendor",
+        "cloud",
+        "version",
+        "managed_cluster_id",
+        "available",
+        "core_worker",
+        "socket_worker",
+        "created_via",
+    )
+    cluster_id = cluster["cluster_id"]
+    # The last 'facts' in the list will contains cluster, operators and acm_metrics information
+    acm_metrics = details["sources"][0]["facts"][-1]["acm_metrics"]
+    assert (
+        len(acm_metrics) > 1
+    ), "Cluster has ACM and no metrics were detected (expected at least one)"
+    assert cluster_id in [
+        x["managed_cluster_id"] for x in acm_metrics
+    ], "Cluster has ACM and its id is not in metrics"
+    for metrics in acm_metrics:
+        assert set(EXPECTED_METRICS_KEYS) == set(list(metrics))
```

Comment:
If you have a managed OpenShift cluster that failed to import, some fields may return as None. I don't know how controlled your provisioned cluster is, but I thought I'd mention it just in case. All keys must be present regardless.

Reply:
Got it @nicolearagao. I'm assuming **the happy path**. The keys will be always there, the values shouldn't be `None`. Let's see how it goes.